contracts/contracts/dapphack.sol

```diff
@@ -134,9 +139,46 @@ contract DappHack is ProjectNFTs {
     }
 
     //create mapping for this
-    modifier NotInTeam() {
+    modifier NotInTeam(address[] memory participants) {
         // need a mapping from address(of the signer) to team as cant fetch a team that is not created
-        // require(s_teams[msg.sender] == 0, "Already in a team");
+
+        for (uint i = 0; i < participants.length; i++) {
+            require(
+                bytes(builderToTeam[participants[i]].name).length == 0,
+                "Already in a team"
+            );
+        }
+
+        require(
+            bytes(builderToTeam[msg.sender].name).length == 0,
+            "Already in a team"
+        );
+        _;
+    }
+
+    modifier DuplicateParticipants(address[] memory participant) {
```

Comment:
many issues here. Modifier is for checks, there should be no state changes there. Moreover, there is no need of duplication mapping. Just make a modifier that takes in array of addresses being added to team and checks if there are repeated addresses in them. I don't get the use and need of duplication mapping hare, it just makes things complex.

Reply:
ok sir